.pre-commit-config.yaml

```diff
@@ -75,7 +75,7 @@ repos:
   hooks:
   - id: mypy
     name: mypy static type checker
-    entry: bash -c 'mypy tools/src/icon4py/tools model/atmosphere/dycore/tests/dycore/stencil_tests' # TODO(egparedes): fix and activate mypy hook for all packages
+    entry: bash -c 'mypy tools/src/icon4py/tools model/testing/src model/atmosphere/dycore/tests/dycore/stencil_tests' # TODO(egparedes): fix and activate mypy hook for all packages
```

Comment:
I enabled mypy on model.testing, which results in most of the changes in this PR.